modules/services/syncthing.nix

```
@@ -10,9 +10,34 @@ with lib;
       enable = mkEnableOption "Syncthing continuous file synchronization";
 
       tray = mkOption {
-        type = types.bool;
-        default = false;
-        description = "Whether to enable QSyncthingTray service.";
+        type = with types;
+          either bool (submodule {
```

```suggestion
          coercedTo bool (value: { default = value; }) (submodule {
```

This should automatically convert the boolean to a submodule.

The logic below will be greatly simplified especially since the command and package are backward compatible.

The test case will need to be updated.

---

Good point :+1: